add per project option to set a custom screenshots path

DamnWidget opened the pull request:
Hi @SolarLune 

This PR adds a per-project option to setup a custom screenshot output directory (if set). It also includes a fallback so in case that the directory does not exists it uses the application path directory as usual.

Regards

SolarLune commented:
Hello! Thanks for the PR submission - this is an interesting feature.

The ProgramSettings variable is for global program settings, not for per-project settings. That should be stored in with the rest of the project information when saving the project (in `Project.Save()`).

I'm also not sure about the use of a `ValidatePath()` function that only is used once in the codebase - that could be done in the place where you're using it.

DamnWidget commented:
I am not sure why but in my mind `programSettings` made a really good "per project" settings, will fix it as soon as I have a gap, will also inline the validation.

DamnWidget commented:
Moved the setting to the project configuration and added a line to persist it on the project `.plan` file, also inlined the validation into `main.go` and removed the Validation function from `common.go` 

Let me know if you want any more changes

SolarLune commented:
OK, this looks good now. Thanks for the assistance!
